content/guides/digging-deeper/route-model-binding.md

```diff
@@ -0,0 +1,279 @@
+---
+summary:  Bind the route parameters with Lucid models and automatically query the database
+---
+
+AdonisJS provides a powerful route model binding feature, which allows you to bind the route parameters with Lucid models and automatically query the database.
+
+The package must be installed and configured separately. You can install it by running the following command.
+
+:::div{class="setup"}
+
+:::codegroup
+
+```sh
+// title: Install
+npm i @adonisjs/route-model-binding
+```
+
+```sh
+// title: Configure
+node ace configure @adonisjs/route-model-binding
+
+# UPDATE: tsconfig.json { types += "@adonisjs/route-model-binding/build/adonis-typings" }
+# UPDATE: .adonisrc.json { providers += "@adonisjs/route-model-binding/build/providers/RmbProvider" }
+```
+
+```ts
+// title: Register middleware
+/**
+ * Make sure to add the following global middleware inside
+ * the start/kernel.ts file
+ */
+Server.middleware.register([
+  // ...other middleware
+  () => import('@ioc:Adonis/Addons/RmbMiddleware'),
+])
+```
+ 
+:::
+
+
+:::div{class="features"}
+
+- Works with all Database drivers
+- Customizable lookup logic
+
+&nbsp;
+
+- [View on npm](https://www.npmjs.com/package/@adonisjs/route-model-binding)
+- [View on GitHub](https://github.com/adonisjs/route-model-binding)
+
+:::
+
+## Example
+
+Route model binding is a neat way to remove one-liner Lucid queries from your codebase and use conventions to query the database during HTTP requests.
+
+In the following example, we connect the route params `:post` and `:comments` with the arguments accepted by the `show` method.
+
+- The value of the first param from the URL will be used to query the first typed hinted model on the `show` method (i.e., Post).
+- Similarly, the value of the second param will be used to query the second typed hinted model (i.e., Comment).
+
+:::note
+The params and models are connected using the order they appear and not the name. This is because TypeScript decorators have no way to know the names of the arguments accepted by a method.
```

Comment:
```suggestion
The params and models are connected using the order they appear and not the name. This is because TypeScript decorators cannot know the names of the arguments accepted by a method.
```